### ISSUE TITLE
Bug in method "clearAll" AWSCognitoIdentityUserPool

### ISSUE THREAD

**Issue by Shahmatist3** [aws/aws-sdk-ios#476]
Hi! Can you please check if the method “clearAll” in class “AWSCognitoIdentityUserPool.m” (line) 180 works correctly?

 In your code you have variable 

`static AWSUICKeyChainStore *keychain = nil; `

and you use this variable with nil in method “clearAll” 

```
- (void) clearAll {
    NSArray *keys = keychain.allKeys;
    NSString *keyChainPrefix = [NSString stringWithFormat:@"%@.", self.userPoolConfiguration.clientId];
    for (NSString *key in keys) {
        if([key hasPrefix:keyChainPrefix]){
            [keychain removeItemForKey:key];
        }
    }
}

```

in this case variable “keys” will be always empty array. Thank you!

**Comment by behrooziAWS** [comment]
keychain is setup on line 115 in the init method:
`_keychain = [AWSUICKeyChainStore keyChainStoreWithService:[NSString stringWithFormat:@"%@.%@", [NSBundle mainBundle].bundleIdentifier, [AWSCognitoIdentityUserPool class]]];`

If you are having issues with the keychain have you turned on the Enable Keychain Sharing capability?

**Comment by Shahmatist3** [comment]
Sorry, maybe I missed some piece of code but

` _keychain = [AWSUICKeyChainStore keyChainStoreWithService:[NSString stringWithFormat:@"%@.%@", [NSBundle mainBundle].bundleIdentifier, [AWSCognitoIdentityUserPool class]]]; 
`
where 
`_keychain`  is a property 
and 
`keychain` is a static variable 

I didn’t find a piece of code where to variable `keychain` (not property) we put some value. 
I tried to put a breakpoint in method clearAll and variable “`keys`” (`keychain.allKeys`) is always empty.
In method `clearAll` i replaced `keychain.allKeys` with `_keychain.allKeys`

**Comment by behrooziAWS** [comment]
Ah! You are correct, will fix in the next release.

**Comment by karthiksaligrama** [comment]
fixed with 2.4.10
